flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/DeclarativeSchedulerBuilder.java

```diff
@@ -0,0 +1,181 @@
+            final ScheduledExecutorService futureExecutor) {
+        this.futureExecutor = futureExecutor;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setUserCodeLoader(final ClassLoader userCodeLoader) {
+        this.userCodeLoader = userCodeLoader;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setCheckpointRecoveryFactory(
+            final CheckpointRecoveryFactory checkpointRecoveryFactory) {
+        this.checkpointRecoveryFactory = checkpointRecoveryFactory;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setRpcTimeout(final Time rpcTimeout) {
+        this.rpcTimeout = rpcTimeout;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setBlobWriter(final BlobWriter blobWriter) {
+        this.blobWriter = blobWriter;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setJobManagerJobMetricGroup(
+            final JobManagerJobMetricGroup jobManagerJobMetricGroup) {
+        this.jobManagerJobMetricGroup = jobManagerJobMetricGroup;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setShuffleMaster(final ShuffleMaster<?> shuffleMaster) {
+        this.shuffleMaster = shuffleMaster;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setPartitionTracker(
+            final JobMasterPartitionTracker partitionTracker) {
+        this.partitionTracker = partitionTracker;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setDeclarativeSlotPool(
+            DeclarativeSlotPool declarativeSlotPool) {
+        this.declarativeSlotPool = declarativeSlotPool;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setRestartBackoffTimeStrategy(
+            final RestartBackoffTimeStrategy restartBackoffTimeStrategy) {
+        this.restartBackoffTimeStrategy = restartBackoffTimeStrategy;
+        return this;
+    }
+
+    public DeclarativeSchedulerBuilder setJobStatusListener(JobStatusListener jobStatusListener) {
+        this.jobStatusListener = jobStatusListener;
+        return this;
+    }
+
+    public DeclarativeScheduler build() throws Exception {
+        return new DeclarativeScheduler(
+                jobGraph,
+                jobMasterConfiguration,
+                declarativeSlotPool,
+                futureExecutor,
+                ioExecutor,
+                userCodeLoader,
+                checkpointRecoveryFactory,
+                rpcTimeout,
+                blobWriter,
+                jobManagerJobMetricGroup,
+                shuffleMaster,
+                partitionTracker,
+                restartBackoffTimeStrategy,
+                new DefaultExecutionDeploymentTracker(),
+                System.currentTimeMillis(),
+                mainThreadExecutor,
+                error ->
+                        FatalExitExceptionHandler.INSTANCE.uncaughtException(
+                                Thread.currentThread(), error),
```

Comment:
I guess we would want to make the fatal error handler also configurable in order to test the `runIfState` method properly.